packages/compartment-mapper/README.md

```diff
@@ -0,0 +1,534 @@
+# Compartment mapper
+
+The compartment mapper builds _compartment maps_ for Node.js style
+applications, finding their dependencies and describing how to create
+[Compartments][] for each package in the application.
+
+Creating a compartment map for a Node.js application allows us to harness
+the SES module loader to encapsulate each dependency and grant the least
+necessary authority to each third-party package, mitigating prototype pollution
+attacks and some supply chain attacks.
+Since most Node.js packages do not modify objects in global scope,
+many libraries and applications work in Compartments without modification.
+
+The `importLocation` function runs a compartmentalized application off the file
+system.
+The `endowments` are properties to add to the `globalThis` in the global scope
+of the application's main package compartment.
+The `modules` are built-in modules to grant the application's main package
+compartment.
+
+```js
+import fs from "fs";
+import { importLocation } from "@agoric/compartment-mapper";
+
+// ...
+
+const modules = { fs };
+const endowments = { console };
+
+const read = async location =>
+  fs.promises.readFile(new URL(location).pathname);
+
+const { namespace } = await importLocation(
+  read,
+  moduleLocation,
+  endowments,
+  modules
+);
+```
+
+The compartment mapper does nothing to arrange for the realm to be frozen.
+The application using the compartment mapper is responsible for applying the
+[SES] shim (if necessary) and calling `lockdown` to freeze the realm (if
+necessary).
+The compartment mapper is also not coupled specifically to Node.js IO and does
+not import any powerful modules like `fs`.
+The user must provide `read` and `write` functions from whatever IO powers they
+have.
+
+> TODO
+>
+> A future version will allow application authors to distribute
+> their global endowments and granted built-in modules to third-party
```

Comment:
Maybe say "their choices of global endowments..", I initially read this as the endowment objects themselves were being serialized somehow, but I think we're talking about manifests which contain string names of endowments/modules.